src/server/conn_context.cc

```diff
@@ -119,6 +119,13 @@ void ConnectionContext::ChangeMonitor(bool start) {
   EnableMonitoring(start);
 }
 
+ConnectionState::ClientTracking& ConnectionContext::ClientTrackingInfo() {
+  if (parent_cntx_) {
+    return parent_cntx_->conn_state.tracking_info_;
```

Comment:
That;s for squashing :)

Reply:
If you access conn_state, don't make it a function on conn_cntx

then you can just use conn_state, you can make it mutable or add a new member like `conn`

https://github.com/dragonflydb/dragonfly/blob/a95419b0c4eaa524347bf7d0a47349f1be07475a/src/server/main_service.cc#L214-L215